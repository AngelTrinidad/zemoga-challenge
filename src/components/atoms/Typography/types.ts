export enum Variant {
  xxl = 'xxl',
  xl = 'xl',
  lg = 'lg',
  normal = 'normal',
  sm = 'sm',
}

export type TypographyVariant = keyof typeof Variant;
