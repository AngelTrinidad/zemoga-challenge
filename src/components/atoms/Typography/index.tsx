import React, {FC} from 'react';
import {TextProps, TextStyle} from 'react-native';

import {TypographyVariant, Variant} from './types';

import Text from '~components/atoms/Text';
import {useTheme} from '~contexts';

export type {TypographyVariant};

interface Props extends TextProps {
  variant?: TypographyVariant;
  color?: 'primary' | 'secondary';
  textAlign?: 'center' | 'left' | 'right';
}

const VARIANT_STYLES: {[key: string]: TextStyle} = {
  [Variant.xxl]: {
    fontSize: 24,
  },
  [Variant.xl]: {
    fontSize: 20,
  },
  [Variant.lg]: {
    fontSize: 16,
  },
  [Variant.normal]: {
    fontSize: 14,
  },
  [Variant.sm]: {
    fontSize: 12,
  },
};

const Typography: FC<Props> = ({
  children,
  style,
  textAlign,
  variant = 'normal',
  color = 'primary',
  ...props
}) => {
  const {
    colors: {text},
  } = useTheme();

  const textStyle: TextStyle = {
    ...VARIANT_STYLES[variant],
    color: text[color],
    textAlign,
  };

  return (
    <Text style={[textStyle, style]} {...props}>
      {children}
    </Text>
  );
};

export default Typography;
