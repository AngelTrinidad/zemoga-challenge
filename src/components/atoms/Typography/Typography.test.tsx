import React from 'react';

import {render} from '@testing-library/react-native';

import Typography from '.';

describe('<Typography />', () => {
  let component: any;

  beforeEach(() => {
    component = render(<Typography>Hello</Typography>);
  });

  it('should renders correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has the same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
