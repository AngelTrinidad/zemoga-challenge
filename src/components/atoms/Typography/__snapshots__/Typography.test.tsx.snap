// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should has the same snapshot 1`] = `
<Text
  style={
    {
      "color": "black",
      "fontFamily": "Metropolis Bold",
      "fontSize": 17.2,
      "fontWeight": "800",
    }
  }
>
  Hello
</Text>
`;
